features/environment.py

```diff
@@ -17,8 +18,18 @@
 
 ALL_SUPPORTED_SERIES = ["bionic", "focal", "xenial"]
 
-DAILY_PPA = "http://ppa.launchpad.net/ua-client/daily/ubuntu"
-DAILY_PPA_KEYID = "6E34E7116C0BC933"
+
+class InstallationSource(Enum):
+    ARCHIVE = "archive"
+    LOCAL = "local"
+    DAILY = "daily"
+    STAGING = "staging"
+    PROPOSED = "proposed"
+    CUSTOM = "custom"
```

Comment:
do we need a stable ppa option?

Reply:
we can always use custom if not

Reply:
I didn't add it because of the low (if any) usage of our stable PPA. If you [check the files](https://launchpad.net/~ua-client/+archive/ubuntu/stable), we are still 27.7 there, and 27.8 has been around for a while. We can use custom, yes.

However, it would be straightforward to include it anyway if you think that will be useful.

Reply:
The fact that our stable PPA is out of date isn't a good thing and shouldn't be a reason not to include it here :upside_down_face: 

It was useful recently for the interim 27.1 that was only released to the stable ppa, so I'm leaning towards including the `stable` target if it's easy enough to add.

Reply:
Sure will do

Reply:
done